pkg/core/children.go

```diff
@@ -199,19 +198,19 @@ func (h *Handler) getObject(namespace, name string, metadata configMetadata, obj
 
 // getExistingChildren returns a list of all Secrets and ConfigMaps that are
 // owned by the Deployment instance
-func (h *Handler) getExistingChildren(obj *appsv1.Deployment) ([]Object, error) {
+func (h *Handler) getExistingChildren(obj podController) ([]Object, error) {
 	inNamespace := client.InNamespace(obj.GetNamespace())
 
 	// List all ConfigMaps in the Deployment's namespace
 	configMaps := &corev1.ConfigMapList{}
-	err := h.List(context.TODO(), configMaps, inNamespace)
+	err := h.List(context.TODO(), inNamespace, configMaps)
```

Comment:
@JoelSpeed I've rebased. I did notice a possible error from a previous commit here.
List is defined as the following
```List(ctx context.Context, opts *ListOptions, list runtime.Object)```

`inNamespace` is of type `*ListOptions`

Reply:
We are currently running a fork of controller-runtime whilst waiting for a bug fix to be merged, so we are ahead of the latest release which includes a bunch of breaking changes including changing the signature of the `List` method to `List(ctx context.Context, list runtime.Object, opts ...ListOptionFunc)` where `inNamespace` is now a `ListOptionFunc`

Good spot though 😄